packages/flame/lib/src/game/flame_game.dart

```diff
@@ -105,12 +93,42 @@ class FlameGame extends Component with Game {
   @override
   @mustCallSuper
   void onGameResize(Vector2 canvasSize) {
+    if (!isMounted) {
+      // TODO(st-pasha): remove this hack, which is for test purposes only
+      setMounted();
+    }
     camera.handleResize(canvasSize);
     super.onGameResize(canvasSize); // Game.onGameResize
     // [onGameResize] is declared both in [Component] and in [Game]. Since
     // there is no way to explicitly call the [Component]'s implementation,
     // we propagate the event to [FlameGame]'s children manually.
-    children.forEach((child) => child.onGameResize(canvasSize));
+    handleResize(canvasSize);
+  }
+
+  /// Ensure that all pending tree operations finish.
+  ///
+  /// This is mainly intended for testing purposes: awaiting on this future
+  /// ensures that the game is fully loaded, and that all pending operations
+  /// of adding the components into the tree are fully materialized.
+  ///
+  /// Warning: awaiting on a game that was not fully connected will result in an
+  /// infinite loop. For example, this could occur if you run `x.add(y)` but
+  /// then forget to mount `x` into the game.
+  Future<void> ready() async {
+    var repeat = true;
+    while (repeat) {
```

Comment:
I am failing to see how this will ensure that the game will be loaded once this future is completed.

`processPendingLifecycleEvents` do call load from children but it don't await for their completion.

Can you shed some light here?

Reply:
`processChildrenQueue` method removes components from the queue only after they finished loading+mounting. Thus, while a component is loading, the getter `.hasPendingLifecycleEvents` will be returning `true`.

Then, in each iteration of this loop we check if there are any queues that has pending lifecycle events. If such queues are found, we make a short async break and start polling all the queues again.

Reply:
Is the loop used here? It looks like it is set to `repeat = false` in the first iteration, and never set to true?

Reply:
Yes, it's set to true below in line 126 whenever a child says that it still has some pending lifecycle events.